text/001x-battery_improvements.md

```diff
@@ -0,0 +1,293 @@
+        <description>Battery firmware is not compatible with current autopilot firmware.</description>
+      </entry>
+      <entry value="131072" name="MAV_BATTERY_STATUS_FLAGS_FAULT_INCOMPATIBLE_CELLS_CONFIGURATION">
+        <description>Battery is not compatible due to cell configuration (e.g. 5s1p when vehicle requires 6s).</description>
+      </entry>
+      <entry value="262144" name="MAV_BATTERY_STATUS_FLAGS_CAPACITY_RELATIVE_TO_FULL">
+        <description>
+          Battery capacity_consumed and capacity_remaining values are relative to a full battery (they sum to the total capacity of the battery).
+          This flag would be set for a smart battery that can accurately determine its remaining charge across vehicle reboots and discharge/recharge cycles.
+          If unset the capacity_consumed indicates the consumption since vehicle power-on, as measured using a power monitor. The capacity_remaining, if provided, indicates the estimated remaining capacity on the assumption that the battery was full on vehicle boot.
+          If unset a GCS is recommended to advise that users fully charge the battery on power on.
+        </description>
+      </entry>
+      <entry value="4,294,967,295" name="MAV_BATTERY_STATUS_FLAGS_EXTENDED">
+        <description>Reserved (not used). If set, this will indicate that an additional status field exists for higher status values.</description>
+      </entry>
+    </enum>
+```
+
+The message is heavily based on [BATTERY_STATUS](https://mavlink.io/en/messages/common.html#BATTERY_STATUS) but:
+- removes the cell voltage arrays: `voltages` and `voltages_ext`
+- adds `voltage`, the total cell voltage. This is a uint32_t to allow batteries with more than 65V.
+- `current_consumed` has changed to capacity_consumed, and from an `int32_t` to a `uint32_t`.
+  This future proofs from 32767 mAh limit to 4294967295 mAh 
+- removes `energy_consumed`, which essentially duplicates `capacity_consumed` for most purposes, and `capacity_consumed` in mAh is nearly ubiquitous.
+- adds `capacity_remaining`(mAh).
+  - This allows a GCS to more accurately determine available current and remaining time than inferring from the `capacity_consumed` and `percent_remaining`.
+  - This will be reliable for smart batteries.
+  - A flag has been added to indicate whether this is calculated from an assumed or known-full battery.
+    A GCS can use this to prompt that batteries be fully charged for power modules.
+- Note that with capacity consumed and remaining, you have the full capacity and you could calculate the percentage remaining.
+  However percentage remaining is supplied anyway, as the other values are optional, and this is actually the one value that most users really want.
+- change `current` from a `int16_t` (cA) to a `int32_t` (mA). Maximum size was previously 327.67A, which is not large enough to be future proof. New value gives up to 2,147,483A. The value is positive when discharging, and negative when charging.
```

Comment:
```suggestion
- adds `voltage`, the total cell voltage. This is a float to allow batteries with more than 65V and also very small voltage changes.
- `current_consumed` has changed to capacity_consumed, and from an `int32_t` to a `float` in A.
  This future proofs from 32767 mAh limit and allows very large and very small values.
- removes `energy_consumed`, which essentially duplicates `capacity_consumed` for most purposes, and `capacity_consumed` in mAh is nearly ubiquitous.
- adds `capacity_remaining`(Ah) as a `float`.
  - This allows a GCS to more accurately determine available current and remaining time than inferring from the `capacity_consumed` and `percent_remaining`.
  - This will be reliable for smart batteries.
  - A flag has been added to indicate whether this is calculated from an assumed or known-full battery.
    A GCS can use this to prompt that batteries be fully charged for power modules.
- Note that with capacity consumed and remaining, you have the full capacity and you could calculate the percentage remaining.
  However percentage remaining is supplied anyway, as the other values are optional, and this is actually the one value that most users really want.
- change `current` from a `int16_t` (cA) to a `float` (A). Maximum size was previously 327.67A, which is not large enough to be future proof. New value gives full range and both very small and very high values.
```